inference-engine/thirdparty/clDNN/src/include/reorder_inst.h

```diff
@@ -33,11 +35,19 @@ struct typed_program_node<reorder> : public typed_program_node_base<reorder> {
     void requires_reinterpret(bool val) { req_reinterpr = (optimized && val); }
 
     void set_input_offset(tensor const& io) { input_offset = io; }
+    void set_input_layout(layout const& lo) { input_layout = lo; }
```

Comment:
Why is it needed? Can't you just use `input().get_output_layout()`

Reply:
This is becuase, when the get_fused_params is called, the connection b/w reorder and input() is removed, so there is no more inputs to reorder. Thus, it was needed to record the original input layout when the reorder is fused.